File: pages/login_page.py
from playwright.sync_api import Page
from utils.helpers import wait_for_2

class LoginPage:
    def __init__(self, page: Page):
        self.page = page
        self.user_inp = "input[data-test='username']"
        self.password_inp = "input[data-test='password']"
        self.login_btn = "input[data-test='login-button']"
        self.error_msg = "[data-test='error']"

    def goto(self, base_url: str):
        self.page.goto(base_url)

    def login(self, username, password):
        self.page.fill(self.user_inp, username)
        self.page.fill(self.password_inp, password)
        wait_for_2()
        self.page.click(self.login_btn)

    def error_message(self):
        if self.error_message:
            return self.error_message.text_content()
        return ""

File: DavidRodriguez/pages/login_page.js

const { waitFor } = require('../utils/helpers');

class LoginPage {
  /**
   * @param {import('@playwright/test').Page} page - Objeto Page que Playwright inyecta en cada test
   */
  constructor(page) {
    this.page = page;// Guardamos la referencia a la página para usarla en los métodos
    // Selectores CSS para los elementos de la página de login
    this.userInp = page.locator("input[data-test='username']");
    this.passwordInp = page.locator("input[data-test='password']");
    this.loginBtn = page.locator("input[data-test='login-button']");
    this.errorMsg = page.locator("[data-test='error']");
  }

  async goto(baseUrl) {
    await this.page.goto(baseUrl);// Abre la página de login
  }

  async login(username, password) {
    await this.userInp.fill(username);
    await this.passwordInp.fill(password);
    await waitFor(this.page, 5); // espera 5 segundos
    await this.loginBtn.click();
  }

  async getErrorMessage() {
    if (await this.errorMsg.isVisible()) {// Verifica si el mensaje de error está visible
      return await this.errorMsg.textContent();// Devuelve el texto del mensaje
    }
    return '';// Si no hay mensaje visible, devuelve cadena vacía
  }
}

module.exports = { LoginPage };